ui/discovery/discovery.go

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discovery
+
+import (
+	"fmt"
+	"sync"
+)
+
+// LANDiscovery provides local network discovery service for Mysterium Node UI.
+type LANDiscovery interface {
+	Start() error
+	Stop() error
+}
+
+type multiDiscovery struct {
+	ssdp    LANDiscovery
+	bonjour LANDiscovery
+	lock    sync.Mutex
+}
+
+// NewLANDiscoveryService creates SSDP and Bonjour services for LAN discovery.
+func NewLANDiscoveryService(uiPort int) *multiDiscovery {
+	outIP := getOutboundIP()
+	url := fmt.Sprintf("http://%s:%d/", outIP, uiPort)
+	return &multiDiscovery{
+		ssdp:    newSSDPServer(url),
+		bonjour: newBonjourServer(uiPort),
+	}
+}
+
+func (md *multiDiscovery) Start() error {
+	if err := md.bonjour.Start(); err != nil {
```

Comment:
Why 2 levels of abstraction? Is Bounjour just for OSX and will have more adapters?

Reply:
Built-in UI has a single object - `discovery`. But we have to implement both SSDP and Bonjour under the hood to support Linux, Mac OS, and Windows.